moveit_planners/pilz_industrial_motion_planner/src/command_list_manager.cpp

```diff
@@ -239,7 +239,11 @@ CommandListManager::solveSequenceItems(const planning_scene::PlanningSceneConstP
     setStartState(motion_plan_responses, req.group_name, req.start_state);
 
     planning_interface::MotionPlanResponse res;
-    planning_pipeline->generatePlan(planning_scene, req, res);
+    if (!planning_pipeline->generatePlan(planning_scene, req, res))
+    {
+      RCLCPP_ERROR(LOGGER, "Generating a plan with planning pipeline failed.");
+      res.error_code.val = moveit_msgs::msg::MoveItErrorCodes::FAILURE;
```

Comment:
Does `generatePlan` already not assign an error code? Just making sure we are not overriding it with the generic FAILURE error code here

Reply:
yes, technically some adapter or the planner should put its own error message into the result.

Reply:
I think it is possible for the chain to fail before anything is written to the error code. E.g. if the chain throws an [exception early](https://github.com/ros-planning/moveit2/blob/095b8ae9f6e3f6218a2f32dd1fbe33a843b085a2/moveit_ros/planning/planning_pipeline/src/planning_pipeline.cpp#L233). That is also the only case when generatePlan returns false